worker.py

```diff
@@ -0,0 +1,173 @@
+import logging
+import sys
+import os
+from base_util import validate_config
+from dane import Document, Task, Result
+from dane.base_classes import base_worker
+from dane.config import cfg
+from dane.provenance import Provenance
+from models import CallbackResponse
+from io_util import (
+    fetch_input_s3_uri,
+    source_id_from_s3_uri,
+    get_s3_output_file_uri,
+)
+from pika.exceptions import ChannelClosedByBroker  # type: ignore
+import main_data_processor
+
+
+logger = logging.getLogger()
+
+
+class WhisperASRWorker(base_worker):
+    def __init__(self, config):
+        logger.info(config)
+
+        if not validate_config(config, not self.UNIT_TESTING):
+            logger.error("Invalid config, quitting")
+            sys.exit()
+
+        self.__queue_name = "QUEUE_NAME"
+        self.__binding_key = "#.BINDING_KEY"
+        self.__depends_on = (
+            list(config.DANE_DEPENDENCIES) if "DANE_DEPENDENCIES" in config else []
+        )
+
+        super().__init__(
+            self.__queue_name,
+            self.__binding_key,
+            config,
+            self.__depends_on,
+            auto_connect=not self.UNIT_TESTING,
+            no_api=self.UNIT_TESTING,
+        )
+
+        # NOTE: cannot be automatically filled, because no git client is present
+        if not self.generator:
+            logger.info("Generator was None, creating it now")
+            self.generator = {
+                "id": "dane-whisper-asr-worker",
+                "type": "dane-whisper-asr-worker",
+                "name": "ASR using Whisper",
+                "homepage": "https://github.com/beeldengeluid/dane-whisper-asr-worker",
+            }
+
+    """----------------------------------INTERACTION WITH DANE SERVER ---------------------------------"""
+
+    # DANE callback function, called whenever there is a job for this worker
+    def callback(self, task: Task, doc: Document) -> CallbackResponse:
+        logger.info("Receiving a task from the DANE server!")
+        logger.info(task)
+        logger.info(doc)
+
+        # fetch s3 uri of input data:
+        s3_uri = fetch_input_s3_uri(self.handler, doc)
+
+        # now run the main process!
+        processing_result, full_provenance_chain = main_data_processor.run(s3_uri)
+
+        # if results are fine, save something to the DANE index
+        if processing_result.get("state", 500) == 200:
+            logger.info(
+                "applying IO on output went well, now finally saving to DANE index"
+            )
+            self.save_to_dane_index(
+                doc,
+                task,
+                get_s3_output_file_uri(source_id_from_s3_uri(s3_uri)),
+                provenance=full_provenance_chain,
+            )
+        return processing_result
+
+    # TODO adapt
```

Comment:
I see that this TODO is still in the example_worker, but we won't be doing anything here - the output will be written to S3 rather than DANE, and reporting back to DANE itself is already covered by this bit of code.
I'll also remove it in the example worker!